backend/src/main/java/com/yohoard/notes/TagService.java

```diff
@@ -0,0 +1,39 @@
+package com.yohoard.notes;
+
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class TagService {
+
+    private final TagRepository tagRepository;
+
+    public TagService(TagRepository tagRepository) {
+        this.tagRepository = tagRepository;
+    }
+
+    public List<Tag> getOrCreate(List<String> names) {
+        List<Tag> existingTags = tagRepository.findByNameIn(names);
+
+        Set<String> existingNames = existingTags.stream()
+                .map(Tag::getName)
+                .collect(Collectors.toSet());
+
+        List<Tag> newTags = names.stream()
+                .filter(name -> !existingNames.contains(name))
```

Comment:
Ta linijka mnie przekonuje, że jeśli miałoby się dać dodać kilka tagów za jednym zamachem to tym bardziej set. Logika jaką chcesz napisać to to że przyjmujesz jakiś zbiór name, wyciągasz zbiór tego co masz w bazie, a potem porównujesz i dla różnicy tworzysz tagi, których nie ma w bazie :P